grundeinkommensbuero/backend
# Rewrote auth using /sign-in endpoint

Resolves: https://expedition-grundeinkommen.monday.com/boards/853311751/pulses/1264619042

The backend was rewritten and now uses a separate /users/sign-in endpoint, which creates a new login code, saves it in the cognito record and then emails it to the user. Therefore in the frontend the whole order of steps is skewed. The new Code is heavily influenced by https://tschoffelen.medium.com/serverless-magic-links-with-aws-cognito-92eff1351123. A lot of the code is basically just copied (especially in the backend). 

The login code is now valid for twenty minutes, but this can easily be changed to a different value. 

Old: 
1) Show Email Form
2) Call Auth.signIn() 
3) Show Login Code Form
4) Call Auth.sendCustomChallengeAnswer()

New:
1) Show Email Form
2) Call /users/sign-in endpoint
3) Show Login Code Form
4) Call Auth.signIn(), which does not do a lot except for initializing the Cognito Flow
5) Call Auth.sendCustomChallengeAnswer()

To test: 
This pr can juste be tested by testing the frontend. Maybe also look over the code and check if something seems funky. 

- Sign in with existing user
- Sign in with a user which does not exist
- Sign up with existing user
- Sign up with new user
